File: src/main/java/VetCare/Back/domain/entities/Veterinario.java
package VetCare.Back.domain.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.Collection;
import java.util.List;

@Entity
@Table(name = "veterinarios")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Veterinario implements UserDetails {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nome;
    @Column(unique = true)
    private String crmv;
    private String especialidade;
    private String telefone;
    @Column(unique = true)
    private String email;
    private String senha;

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return List.of(new SimpleGrantedAuthority("ROLE_VETERINARIO"));
    }

    @Override
    public String getPassword() {
        return this.senha;
    }

    @Override
    public String getUsername() {
        return this.email;
    }
}

File: front/app/types/veterinario.ts
export interface VeterinarioCadastroRequest {
  nome: string;
  crmv: string;
  especialidade?: string;
  telefone?: string;
  email: string;
  senha: string;
}

export interface VeterinarioUpdateRequest {
  nome: string;
  crmv: string;
  especialidade?: string;
  telefone?: string;
  email: string;
  senha?: string;
}

export interface VeterinarioResponse {
  id: number;
  nome: string;
  crmv: string;
  especialidade: string;
  telefone: string;
  email: string;
}
